### ISSUE TITLE
Growth hacking

### ISSUE THREAD

**Issue by rnystrom** [rnystrom/Freetime#192]
I [shared a thread](https://twitter.com/_ryannystrom/status/890680066870571008) about current download stats. Pretty crummy! I'm building this app for myself first, but want to learn more about App Store marketing. I've written apps that have been featured before, but I figured since this is free (and I wont stop tweeting about it) it'd take off to a thousand downloads a week (give or take).

Opening this issue to document ideas about ways to improve reach/downloads:

- More targeted keywords
- Rename in App Store
- Prompt for reviews (maybe after marking a few threads as read)
- FB ads (I have a credit)
- [Blogging](http://whoisryannystrom.com/)
- Static landing page/site (SEO)
- Add a "Posted with Freetime" (+link) when commenting
- Post in HN/Dribbble/DN (what else?)

The challenge: w/e we do, it should be (basically) free.

**Comment by rnystrom** [comment]
What tools exist nowadays to check app keyword ranking? There used to be tools to min/max and lint your keywords.

**Comment by flovilmart** [comment]
We use sensor tower and app Annie.
Reviews on blogs/articles can get the base started. A github shield, could interesting (Maintained with Freetime).

**Comment by flovilmart** [comment]
Project on open collective + patreon, not for the money but for the traffic.

**Comment by rnystrom** [comment]
@flovilmart those are so good!

**Comment by rnystrom** [comment]
App Store title:
```
Freetime - GitHub open source project manager
```

Keywords:

```
github, open, source, notification, manager, git, pull, request, issue
```

**Comment by flovilmart** [comment]
Perhaps focusing on “github” keyword ASO is enough, the space is crowded but poorly, most app have no ranking, this make a big diff. 

In terms of keyword that should be good. 

Be careful with the byline, put it in the new field in AppStore or you may face a rejection.

**Comment by BasThomas** [comment]
I had some thoughts on this as well after reading your thread. 

On reviews: I would put this in settings. No popups; they suck. 
On posting on sites: Reddit? Not sure what the best subreddit would be. What about Product Hunt?

* Add App Store badge+link in Readme
* Ask for GH stars in some way? Not sure how / where
* Somehow promote the accessibility features might help?

**Comment by Sherlouk** [comment]
Agree with Bas about the traditional popups being really annoying; but I can say that in an app I've published we had an astonishing response to using the new iOS 10.3 ratings prompt (Went from around 500 1 star reviews 😬, to 18,500 reviews with an average of 4.5 stars)

Gotta be careful with that dialog though because can only show it 3 times a year etc

**Comment by flovilmart** [comment]
In that case you do a small 'pre-permission' banner, perhaps once the user achieved something 'meaningful' like posting a first comment, reaching IB Zero, etc... Tying the request for review after a great experience is more likely to yield better result. 

I really the defunct mailbox had those nice pictures when reaching Inbox Zero with a prompt to share.

**Comment by BasThomas** [comment]
I am really curious about asking for ratings after an acting vs putting it in settings and let the user decide. I don't think I've ever wanted to rate an app via popup, not even after a meaningful event. 

On Mailbox: that was so awesome. Could do something like that with the current logo, but that would need some kind of backend etc... might not be worth the effort, especially when it would be such a blatant copy. 😛

**Comment by flovilmart** [comment]
> need some kind of backend

you can probably use unsplash and use a random image every day

> I don't think I've ever wanted to rate an app via popup

Do you use Spark as an email client? from times to times, they inject inline with the mailbox a prompt to rate the app, and they've been doing it for a while so I guess it yields enough interesting results. 
Also on that, a user is more likely to deliver on something you ask for outside the core value of your product (share/rate etc...) if you deliver first your core value. as @Sherlouk mentions, those dialogs deliver, and you can't really waste them, so having a properly timed 'pre-permission' UI, injected into the issues list and not always displayed.

Actually there's nothing much to loose with a flow:

At most once a week -> user has meaningful experience (close issue, comment, discover new feature) -> after action complete and back on home feed -> show Pre-Alert (Hey! Looks like Freetime gets S**T done for you! would you like to write a small review...) -> Act accordingly.

**Comment by shaps80** [comment]
**Rating**
I totally agree with the new integrated ratings dialog. Its very simple and non-intrusive (not to mention users can disable it completely for an app in Settings) and I know personally a lot of companies have seen a huge uptake in getting more interaction from their users by switching to it. 

In terms of where and when to show it, most of you have covered the points above but just to throw another option in there, what about when the unread notifications reaches empty. Instead of having just a bouncing icon and some text. You can add something like:

_'Enjoying Freetime? Rate it on the AppStore'_ – with a button below it.

Simple, non-intrusive.

**Sharing**

1. Perhaps adding an invite button to send out a link to the AppStore build to everyone or a specific person in an issue?

2. You could also include a short link/signature in every comment – maybe that's a little overboard but it is a free app, and means the marketing is on a website technically, not in your app.
The user can always delete the signature anyway if they choose. Also could be a setting to disable it.

3. Definitely a way to share the app using the Share Sheet from Settings. Minimum!

**Marketing**

I definitely think a short video (as a gif as well) + screenshots is a great idea. I like the idea of a static single page site as well. At least something for people to share.

Also make a bigger deal of each release. Maybe even hold off on a release a little to ensure there's 2-3 things in the release. Use Twitter hashtags too, #dev etc...

**Comment by flovilmart** [comment]
> You could also include a short link/signature in every comment 

Yeah that would pollute quickly the github repos you maintain, unless it's written in the composition box (and not added on the back of the user (dark pattern))

Perhaps add a label to the issues automatically when replied with it (Replied with Freetime, Closed with Freetime, Merged with Freetime)

That give the benefit also for other users to see that those issues are actively discussed?

**Comment by shaps80** [comment]
I like the label idea, but agree its not at all ideal. I much prefer the sharing options paired with empty notification rate button.

**Comment by flovilmart** [comment]
I guess ill never see those empty notifications, seems that ‘hardcore’ maintainers with 10+ repos can’t follow up ;)

**Comment by rnystrom** [comment]
re: Product Hunt, we already got a [good bump](https://www.producthunt.com/posts/freetime) from that, so I think that option is gone.

@BasThomas 

> Add App Store badge+link in Readme

Perfect + easy, let's do it!

> Ask for GH stars in some way? Not sure how / where

Love it. How's this sound:

- Add a Settings option called "Star on Freetime on GitHub"
- Store a local setting so once its starred, remove the option
- Can use the [addStar](https://developer.github.com/v4/reference/mutation/addstar/) API
  - Requires a GQL id, which might change. Might have to query this repo first, then send the mutation once we have the id.

> Somehow promote the accessibility features might help?

Hmm any examples of apps that do this?

> Fancier tweets for new versions and ask for retweets?

😄 I like this. Even "pls RT" might help.

> Option to share the app with your friends / coworkers / etc in Settings?

Great! Yes!

**Comment by shaps80** [comment]
That's a valid point, mine hasn't been empty for days either. Maybe as @rnystrom previously suggested it could be presented for every few READ's. Like after 5, then another 20, then another 50, etc...

**Comment by flovilmart** [comment]
That's true! Like variable rewards! Hey you're on fire today! That's 20 done!

**Comment by rnystrom** [comment]
Getting a domain, available options so far:

- freetime-app.com
- getfreetimeapp.com

**Comment by shaps80** [comment]
freetime-app.com – good choice

**Comment by rnystrom** [comment]
[freetime-app.com](http://freetime-app.com/) is live!

**Comment by rnystrom** [comment]
Some cool advice on screenshots [here](https://www.appshots.io/app-screenshot-guide). My takeaways:

- I need to use up all 5 screenshots
- Advertise iPad+iPhone on both plats
- **Sell usefulness** not just features

As far as the 5 shots go, I want to show a device w/ a blue background and white text. The shots are:

- "Manage GitHub Notifications" showing inbox
- "Read, Reply, React" showing an issue, highlight reaction feature?
- "Faster Commenting" showing up-close on text action bar
- "Manage Issues & PRs" showing off edit labels, view files
- "iPhone & iPad" showing iPhone and iPad devices w/ screenshots. Maybe the one from the [website](http://freetime-app.com/)

Thoughts?

**Comment by zats** [comment]
Might be cool to record a video walkthrough of a typical flow.
Video autoplays in iOS 11 so it gives opportunity to tell a story before people start looking at screenshots or read description

**Comment by rnystrom** [comment]
@zats yes!! Been meaning to get to this. Might do a super simple one now.

**Comment by BasThomas** [comment]
What about a press kit?

**Comment by Sherlouk** [comment]
Option to share the app can be done via 3D touch on compatible devices, still not against putting that in the Settings page along with the work for starring (#210) - Think if we can get on the GitHub featured list it would definitely help 😅 

What are the current tags/keywords we use? (Likely changed since last comment)

Nice to see that we're benefiting from the App Preview etc

**Comment by rnystrom** [comment]
Good point, here's my current keywords:

- open source
- git
- issue
- client
- codehub
- gitbucket
- ioctocat
- trident
- gitlab
- repository
- gitbucket
- gitter

I'm obviously targeting competition and related apps at the moment. I can't really think of quality keywords when thinking about "github" or "git"...

I'm actually going to open an issue that we create an App Store metadata file in the repo so we can open PRs on App Store info.

**Comment by Sherlouk** [comment]
So genuinely, when I was having a nosey earlier at the rankings on the App Store - I misspelled GitHub (Think I did GiyHub) and was surprised to actually have about 5 entries (not incl. us)

It may not be a bad idea to add a couple misspelled ones!

gooogle.com exists for a reason 😉

**Comment by rnystrom** [comment]
I actually remember reading that allegedly the App Store handles common misspellings for you. Idk how well this works tho. It could be worth dropping some competition that we already rank above w/ misspellings. At least try it out?

https://www.macrumors.com/2013/11/26/app-store-searches-now-compensate-for-typos-and-misspellings/

https://www.apptamin.com/blog/app-store-optimization-aso-app-name-and-keywords/

> Finally, Apple gives some “free” matches on some specific keywords like the keyword “app” or **common misspelling of your brand name.**

Again idk how much I believe this w/out seeing it in practice, ha. I've def misspelled it before and didn't see GitHawk show up... 🤔 

Any suggestions that you think we should include?

**Comment by Sherlouk** [comment]
I mean there's only one way to find out right? Spend a couple minutes trying a whole bunch of different keywords intentionally misspelling some. Write them down (because it's not science otherwise 😉) If you don't see us, add it as a keyword and see in a few days if we appear?

In the past when I've wanted a similar tool I've searched GitHub, it seems the most obvious to try a couple variations of (I have fat fingers, okay?)

**Comment by Sherlouk** [comment]
So just had a bit more of a play round, I searched "GirHub" and it listed about 7 apps (not incl. us), then "GiyHub" and it did list us (and pretty much all the same results as "GitHub"), I did "GirHub" again and it listed everything (again, as if I searched "GitHub").

Apple definitely do some funky stuff...

**Comment by rnystrom** [comment]
Summary of keyword ranking

<img width="538" alt="screen shot 2017-10-14 at 2 44 31 pm" src="https://user-images.githubusercontent.com/739696/31578379-3d4fa0ea-b0ee-11e7-833a-1e20b4f0198f.png">

**Comment by rnystrom** [comment]
Other ideas: searching around for questions/articles about GitHub iOS apps and leaving comments. Ex:

https://www.quora.com/What-is-the-most-functional-and-fully-featured-iOS-app-for-GitHub

**Comment by n3rdankit** [comment]
You can refer this content where you get skills to build a blog...
https://learn.bettercreator.com/s/store/courses/description/5e23de900cf2fc39d2352aaf--blogging-course-by-bettercreator?affCode=I0NOQP
